apps/native-component-list/src/components/HeaderIconButton.tsx

```diff
@@ -1,33 +1,32 @@
 import Ionicons from '@expo/vector-icons/build/Ionicons';
 import * as React from 'react';
-import { TouchableOpacity, View } from 'react-native';
+import { PropsWithChildren } from 'react';
+import { StyleSheet, TouchableOpacity, TouchableOpacityProps, View, ViewProps } from 'react-native';
 
-export function HeaderContainerRight(
-  props: React.ComponentProps<typeof View> & {
-    children?: any;
-  }
-) {
-  return <View {...props} style={[{ paddingRight: 8, flexDirection: 'row' }, props.style]} />;
-}
+const HeaderContainerRight = (props: PropsWithChildren<ViewProps>) => (
+  <View {...props} style={[styles.container, props.style]} />
+);
 
-type Props = {
+type Props = TouchableOpacityProps & {
   color?: string;
-  disabled?: boolean;
   name: string;
-  onPress: () => void;
   size?: number;
 };
 
-export default function HeaderIconButton({
-  color = 'blue',
-  disabled,
-  name,
-  onPress,
-  size = 24,
-}: Props) {
-  return (
-    <TouchableOpacity disabled={disabled} style={{ paddingHorizontal: 12 }} onPress={onPress}>
-      <Ionicons size={size} color={color} name={name as any} />
-    </TouchableOpacity>
-  );
-}
+const HeaderIconButton = ({ color = 'blue', disabled, name, onPress, size = 24 }: Props) => (
+  <TouchableOpacity disabled={disabled} style={styles.iconButton} onPress={onPress}>
+    <Ionicons size={size} color={color} name={name as any} />
+  </TouchableOpacity>
+);
+
+const styles = StyleSheet.create({
+  container: {
+    paddingRight: 8,
+    flexDirection: 'row',
+  },
+  iconButton: {
+    paddingHorizontal: 12,
+  },
+});
+
+export { HeaderContainerRight, HeaderIconButton };
```

Comment:
I believe this one is not so right 🤔 
If we're following the convention: `FileName` ➡️ `what's actually inside the file` then we need to expose default export in this file (as it was previously).
`HeaderIconButton.tsx` suggests that there's a single component inside and it's name is `HeaderIconButton`.
I would suggest considering this namespace/nested-components style here:
```tsx
HeaderIconButton.HeaderContainerRight = HeaderContainerRight;
export default HeaderContainerRight;
```

Reply:
I have tried in there to standardize a bit, since it was the only component which uses `export default` and `export` for two component in same file. The approach you have mentioned seems fine too, but add a bit of complication, what do you think about just extracting `HeaderContainerRight` to the separate file? In most cases I like the one file - one component approach. 🙂